audio/out/ao_coreaudio_utils.c

```diff
@@ -456,7 +456,14 @@ int64_t ca_get_device_latency_us(struct ao *ao, AudioDeviceID device)
         }
     }
 
-    return ca_frames_to_us(ao, latency_frames);
+    double sample_rate = ao->samplerate;
+    OSStatus err = CA_GET_O(device, kAudioDevicePropertyNominalSampleRate, &sample_rate);
```

Comment:
Is it worth doing this once during setup and storing the value? Or do you expect it can change?

Reply:
Hm never mind I think this already only happens once. Misunderstood the context.

Reply:
Yup. In theory we might have to update the values during runtime with listening for changes or so when the device configuration changes, but I don't think that is currently being handled really at this moment, either.

So as such this is explicitly an improvement :) .